authorizations/authorization-ranger/src/main/java/org/apache/gravitino/authorization/ranger/RangerAuthorizationPlugin.java

```diff
@@ -273,8 +273,13 @@ public Boolean onRoleDeleted(Role role) throws AuthorizationPluginException {
       rangerClient.deleteRole(
           rangerHelper.generateGravitinoRoleName(role.name()), rangerAdminName, rangerServiceName);
     } catch (RangerServiceException e) {
-      // Ignore exception to support idempotent operation
-      LOG.warn("Ranger delete role: {} failed!", role, e);
+      if (e.getMessage().contains("No RangerRole found for name")) {
```

Comment:
The `e.getMessage().contains("No RangerRole found for name")` is Not a stable way. I think we can use `rangerClient.getRole()?` to double check it.

Reply:
if role doesn't, `getRole` will return exception. It isn't a good way.

Reply:
Maybe try `e.getStatus()` ?

Reply:
> Maybe try `e.getStatus()` ?

The status is 400. It includes many other errors which shouldn't be ignored.

Reply:
sigh ... why don't they return 404?